### PR TITLE
ENH: report duplicate IDs during DissimilarityMatrix validation

### LINKED ISSUE #813: improve duplicate IDs error message in DissimilarityMatrix
When constructing a `DissimilarityMatrix` with duplicate IDs:

```python
In [1]: from skbio.stats.distance import DissimilarityMatrix

In [2]: dm = DissimilarityMatrix([[0,1],[1,0]], ['a', 'a'])
---------------------------------------------------------------------------
DissimilarityMatrixError                  Traceback (most recent call last)
<ipython-input-2-e7b33aba5668> in <module>()
----> 1 dm = DissimilarityMatrix([[0,1],[1,0]], ['a', 'a'])

/Users/jairideout/dev/scikit-bio/skbio/stats/distance/_base.pyc in __init__(self, data, ids)
    186         ids = tuple(ids)
    187
--> 188         self._validate(data, ids)
    189
    190         self._data = data

/Users/jairideout/dev/scikit-bio/skbio/stats/distance/_base.pyc in _validate(self, data, ids)
    672                                            "diagonal can only contain zeros).")
    673         elif num_ids != len(set(ids)):
--> 674             raise DissimilarityMatrixError("IDs must be unique.")
    675         elif num_ids != data.shape[0]:
    676             raise DissimilarityMatrixError("The number of IDs must match the "

DissimilarityMatrixError: IDs must be unique.
```

It'd be useful to update the error message to include the offending IDs.

### PR DESCRIPTION
**Pull request by jairideout**
Fixes #813.

### CODE REVIEW

File: skbio/stats/distance/_base.py [hunk 1]
```diff
@@ -685,6 +689,11 @@ def _is_id_pair(self, index):
                 len(index) == 2 and
                 all(map(lambda e: isinstance(e, string_types), index)))
 
+    def _find_duplicates(self, iterable):
```
Comment (ElDeveloper) [inline skbio/stats/distance/_base.py]: We recently wrote a function to do this in Qiita, some profiling using IPython seems to show it being faster than this implementation ... just sayin'

```python
In [27]: %timeit h = _find_duplicates(x)
1000 loops, best of 3: 368 µs per loop

In [28]: h
Out[28]: {0, 1, 2, 3, 4, 5, 6, 7, 8, 9}

In [29]: %timeit h = find_repeated(x)
10000 loops, best of 3: 147 µs per loop

In [30]: h
Out[30]: {0, 1, 2, 3, 4, 5, 6, 7, 8, 9}
```

[Code in Qiita](https://github.com/biocore/qiita/blob/01225be1227d788173b43cf6c1b047cc2076bfeb/qiita_db/util.py#L901).

If we put this in scikit-bio, we could import it in Qiita :neckbeard:

Comment (jairideout) [inline skbio/stats/distance/_base.py]: Awesome, thanks! I'll port this function over.

Comment (ElDeveloper) [inline skbio/stats/distance/_base.py]: Awesome, great to hear that!

Thanks!

Yoshiki.

**Comment by jairideout** [comment]
@ElDeveloper I ported the duplicate-finding code from Qiita to `skbio.util.find_duplicates`. Thanks for suggesting this! We're planning on a minor scikit-bio release within the next few days, so this functionality should be available to Qiita soon.

This is ready for review. @ebolyen @gregcaporaso @ElDeveloper is anyone available to take a pass through?

**Comment by ElDeveloper** [comment]
Thanks! Reviewing right now!

**Comment by ElDeveloper** [comment]
:+1:

**Comment by ElDeveloper** [comment]
Thanks for porting the function over, after the release is made, I'll update qiita to import from skbio.

**Comment by jairideout** [comment]
Thanks!!

**Pull request merged**